File: fixtures/typescript/renamed_function_duplicate/right.ts
export function buildInvoiceRows(invoices: Invoice[]): Line[] {
  const closed = invoices.filter((invoice) => invoice.closed);
  const domestic = closed.filter((invoice) => invoice.region === "NA");
  const lines = domestic.map((invoice) => ({ key: invoice.key, value: invoice.total }));
  const kept = lines.filter((line) => line.value > 0);
  return kept;
}

File: fixtures/haskell/renamed_function_duplicate/right.hs
module Invoices where

buildInvoiceRows invoices =
  let active = filter (\invoice -> invoice.amount > 0) invoices
      rows = map (\invoice -> (invoice.id, invoice.amount, invoice.currency)) active
      total = foldl (\acc row -> acc + row.amount) 0 active
  in map
      (\row ->
        if row.amount > total
          then (row.id, row.currency, "large")
          else (row.id, row.currency, "standard"))
      active
